### DISCUSSION TITLE
SLEAP Runs Slow  With Multiple Videos Loaded

### DISCUSSION THREAD

**Discussion by NSGregory** [talmolab/sleap#1772] [Help!] [answered]
Hi,

I find that if I keep the number of videos around 10 or less SLEAP runs very smoothly, but after that number the program starts to bog down significantly especially if I start using videos > 100 MB in size.  For example, my project has 20 videos of 100 - 120 MB each and if I load them all at once SLEAP effectively stops.  My computer has 32 GB RAM which should be able to handle that even if all the videos are loaded at once.  This is happens even if I'm not doing anything, ie, I load up SLEAP and just try to navigate menus the program is very slow.  Is there a known limit to the number of videos, individual file size, or combined file size that we should try to stay under?  I can downsample the videos but would prefer to avoid taking the extra step.

**Comment by talmo** [comment]
Hey @NSGregory,

This definitely shouldn't happen -- we routinely use the labeling GUI with 30+ videos that are much larger.

1. Where are your videos stored? Locally on disk or over a network share?

2. Is there anything being printed in the terminal?

3. What OS/platform are you on?

Cheers,

Talmo

**Reply by NSGregory** [reply]
1. Locally on an SSD
2. Nothing that I have noticed during the slow episodes
3. Windows 11

**Reply by talmo** [reply]
Hmmm, not much to go on then if there are no logs and no obvious issues with the environment...

I really don't think it would affect the sluggishness on the menu and other GUI components, but if you're sure that the culprit is the number of videos, you could try to [reencode the videos](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) so that they're more optimally seekable. We haven't seen it happen before, but in theory it could be due to SLEAP querying the videos for metadata and it being slow on these particular video files (perhaps we're doing it too often and hadn't noticed the slowdown?).

In any case, we need definitely to be able to reproduce the behavior in order to diagnose it and come up with a fix. I appreciate that it's a lot of data, but if you don't mind [sharing the videos with us](https://forms.gle/KebBPdjoHw2WMKmV8), we can try to reproduce the sluggishness on our end.

**Reply by NSGregory** [reply]
I uploaded the videos as you suggested, hopefully there's something useful there.

**Reply by NSGregory** [reply]
I am following up on this since I'm still running into the issue quite frequently.  I've tried exporting the labeled images and starting a new project each time it starts to slow down when adding videos and that helps some, but eventually every project gets bogged down to the point I can't use it any more.  Any thoughts on what I could try to fix it?  My videos are MP4s with file sizes < 100 MB.

**Reply by NSGregory** [reply, marked as answer]
I ran across issue #1058.  Turns out I had 1.5 million predictions in the project.  Clearing them out fixed the issue.  Thanks for your help, luckily it's an easy fix.!
